packages/browser/src/Session.ts

```diff
@@ -238,6 +263,15 @@ export class Session extends EventEmitter {
           await this.logout();
         }, sessionInfo.expirationDate - Date.now());
       }
+    } else {
+      // Silent authentication happens after a refresh, which means there are no
+      // OAuth parames in the current location IRI. It can only succeed if a session
+      // was previously logged in, in which case its ID will be present with a known
+      // identifier in local storage.
+      await silentlyAuthenticate(
+        window.localStorage.getItem(KEY_CURRENT_SESSION),
+        this.clientAuthentication
+      );
```

Comment:
As a follow-up to the suggestion above, this would be become (just a bit more readable and cleaner I think):
```suggestion
      // Check if we have a locally stored session ID...
      const storedSessionID = window.localStorage.getItem(KEY_CURRENT_SESSION);
      // ...if not, then there is no ID token, and so silent authentication cannot happen, but
      // if we do have a stored session ID, attempt to re-authenticate now silently.
      if (storedSessionID !== null) {
        await silentlyAuthenticate(storedSessionID, this.clientAuthentication);
      }
```